consensus/istanbul/validator/weighted.go

```diff
@@ -848,9 +852,13 @@ func (valSet *weightedCouncil) refreshProposers(seed int64, blockNum uint64) {
 	var candidateValsIdx []int // This is a slice which stores index of validator. it is used for shuffling
 
 	for index, val := range valSet.validators {
-		weight := val.Weight()
-		for i := uint64(0); i < weight; i++ {
+		if valSet.chain.Config().Rules(new(big.Int).SetUint64(blockNum)).IsKore {
```

Comment:
Can we use `chainRules` in `refreshProposers()` just like `calcWeight()` does?

Reply:
Yes. Im thinking about it.
Thank you for your review!

Reply:
Reflected in `7d614bf47a583e0eb4f00913fa7826aefdb59a7e`
Thank you!